controllers/eventController.js

```diff
@@ -0,0 +1,96 @@
+import Organizer from "../models/organiser";
+import Event from "../models/event";
+const orgNotFound = new Error("Organizer not found");
+const eventNotFound = new Error("Event not found");
+export async function getEventByDepartment(req, res) {
+  try {
+    const event = await Event.findAll({
+      include: [
+        {
+          model: Organizer,
+          where: {
+            department: {
+              [Sequelize.Op.iLike]: req.params.department,
+            },
+          },
+        },
+      ],
+    });
+    if (!event) {
+      throw eventNotFound;
+    }
+    res.status(200).json(event);
+  } catch (e) {
+    if (e == eventNotFound) {
+      console.log(
+        "No data for event based on department: " + req.params.department
+      );
+      res.sendStatus(404);
+    } else {
+      console.log(e);
+      res.sendStatus(500);
+    }
+  }
+}
+
+export async function getEventByName(req, res) {
+  try {
+    const event = await Event.findAll({
+      where: {
+        eventName: req.params.eventName,
+      },
+    });
+    if (!event) {
+      throw eventNotFound;
+    }
+    res.status(200).json(event);
+  } catch (e) {
+    if (e == eventNotFound) {
+      console.log("No data for event based on name: " + req.params.eventName);
+      res.sendStatus(404);
+    } else {
+      console.log(e);
+      res.sendStatus(500);
+    }
+  }
+}
+
+export async function deleteEvent(req, res) {
+  try {
+    const event = await Event.findOneAndDelete({
+      where: {
+        eventName: req.params.eventName,
```

Comment:
POST method will contain the eventName as a property in res.body instead of req.params.